### ISSUE TITLE
Hidden Field Data Providers - Nothing is submitted

### ISSUE THREAD

**Issue by dar111** [bolt/boltforms#201]
Hi all,

bolt 3.3.6
boltforms 4.1.13
No other extensions installed besides the ones that come with boltforms.

I was trying to set up a new boltform and I wanted to use hidden fields with data like remote ip or random strings. Basically everything is working fine (submitting the form via mail and storing it to the database to a certain contenttype) for regular form types, except the hidden fields with tagged with an event dispatcher. That data is not transmitted and I don't know why. 

I tried to set a label for those fields (instead of "false") but that label is also not transmitted. 

Here is an extract of my boltforms configuration:

`vvk:

    notification:
        enabled: true
        debug: false
        subject: New Form 
        from_name: nachname               # A field name, specified in the fields: section below
        from_email: email             # A field name, specified in the fields: section below
        replyto_name: nachname            # A field name, specified in the fields: section below
        replyto_email: email          # A field name, specified in the fields: section below
        to_name: Receiver              # Must be valid text
        to_email: validmail@example.net # Must be valid email address
    feedback:
        success: Message submission successful
        error: There are errors in the form, please fix before trying to resubmit
    fields:
        nachname:
            type: text
            options:
                required: true
                label: Nachname
                attr:
                    placeholder: Nachname
                constraints: [ NotBlank, { Length: { 'min': 3, 'max': 128 } } ]
        email:
            type: email
            options:
                required: true
                label: E-Mail Adresse
                attr:
                    placeholder: E-Mail Adresse
                constraints: [ NotBlank, Email ]
        random_field:
            type: hidden
            options:
                label: false
            event: 
                name: random_string
                params:
                    length: 12                  # Optional, defaults to 12
        remote_ip:
            type: hidden
            options:
                label: false
            event: 
                name: server_value
                params:
                    key: REMOTE_ADDR
        sent_on:
            type: hidden
            options:
                label: false
            event: 
                name: timestamp
                params:
                    format: '%F %T'
        submit:
            type: submit
            options:
                label: Bestellen
                attr:
                    class: button primary
`

In the _blocks.twig for the mail I added a dump and I only get an array with the two regular form fields.
`{{ dump(data) }}`
`array:2 [
  "nachname" => "Nachname"
  "email" => "mail@example.net"
]
`

My syslog throws shows this exception, but it does not seem related to above mentioned problem, because the exception vanishes when I add a valid redirect url (this reminds me of #200)

`
Critical | exception | contentlink | [BoltForms]: An event dispatcher encountered an exception. 
xxxx\Redirect.php :: 91
`

Would really be grateful if someone could give me a hint or maybe it's a bug?

**Comment by dar111** [comment]
Update: I've tried now every 4.x version and i got the same problem again and again, but with version 3.1.0 it works fine.

**Comment by xphoenyx** [comment]
Same here. I've created a simple contact form, and every time the data from it is submitted, I get a new record in system log:

```
[BoltForms]: An event dispatcher encountered an exception. 
URI: /et/contact
Source: D:\Development\web\project\extensions\vendor\bolt\boltforms\src\Submission\Processor\Redirect.php :: 91
```

Despite this, the script is sending emails successfully.

My form config:

```
templates:
    form: partials/_contactForm.twig
    email: partials/_email.twig
submission:
    ajax: false
notification:
    enabled: true
    debug: false
    debug_address: address@example.com
    debug_smtp: true
    subject: "New message" 
    from_name: Website
    from_email: from_address@example.com
    replyto_email: reply_address@example.com
    replyto_name: Website
    to_name: Test
    to_email: address@example.com
feedback:
    success: "Success"
    error: "Error"
fields:
    name:
        type: text
        options:
            required: true
            constraints: [ NotBlank, { Length: { 'min': 3, 'max': 128 } } ]
    email:
        type: email
        options:
            required: true
            constraints: [ NotBlank, Email ]
    phone:
        type: text
        options:
            required: false
            constraints: [ NotBlank, { Length: { 'min': 6, 'max': 15 } } ]
    subject:
        type: text
        options:
            required: true
            constraints: [ NotBlank, { Length: { 'min': 3, 'max': 30 } } ]
    message:
        type: textarea
        options:
            required: true
    submit:
        type: submit
```

Versions:
Bolt: 3.4.6
BoltForms: 4.1.14

**Comment by rossriley** [comment]
This one should hopefully be fixed in #227

It'll be in the next release which is coming very soon.
